src/ParallelAlgorithms/Interpolation/Actions/InterpolatorReceivePoints.hpp

```diff
@@ -97,37 +99,50 @@ struct ReceivePoints {
                   .infos;
 
           // Add the new target interpolation points at this temporal_id. There
-          // are two conditions that allow us to overwrite the current target
-          // points. Either
+          // are three conditions that allow us to overwrite the current target
+          // points:
           //
           //  1. There are no current target points at the temporal_id, OR
           //  2. There are target points already at this temporal_id, but the
           //     iteration of the new target points is greater than the
           //     iteration of the current target points.
+          //  3. There are target points already at this temporal_id, but the
+          //     iteration of the new target points is equal to the iteration of
+          //     the current target points, and we are retrying interpolation at
+          //     the same iteration.
```

Comment:
This sounds like a message-order race condition.  Tracing through the logic, it looks like if messages arrive in the wrong order you will silently discard the updated one.

Reply:
Yes I believe you're right. I pushed a fixup which I think addresses this. Basically, if the interpolator is sent a new iteration than is currently stored, we just always accept it, rather than what I had before where I checked if it was a reinterpolation or not. Then if we have multiple messages of a new iteration floating around, it won't matter which is received first.

Reply:
That's better, but I think there's still a race if you have to retry twice.  Could you just consider these retries as new iterations so we don't need a new set of logic for them?

Reply:
The `FastFlow` class is in charge of the iterations and has internal logic regarding this, so I don't think trying to externally modify that is a good idea and could cause potential issues.

But yes I believe there still is a race condition. I think I just need to store retry iterations as well. This will all be made a lot easier once we consolidate the horizon finders to just singletons rather than having it as this group+singleton combo.